coalib/parsing/DefaultArgParser.py

```diff
@@ -243,14 +243,26 @@ def default_arg_parser(formatter_class=None):
         '-r', '--relpath', nargs='?', const=True,
         help='return relative paths for files (must be called with --json)')
 
-    outputs_group.add_argument(
+    devtool_exclusive_group = arg_parser.add_mutually_exclusive_group()
+
+    devtool_exclusive_group.add_argument(
         '--debug-bears', nargs='?', const=True,
         help='Enable bear debugging with pdb, that can help to identify and'
         ' correct errors in bear code. Steps into bear code as soon as being'
         ' executed. To specify which bears to debug, supply bear names as'
         ' additional arguments. If used without arguments, all bears specified'
         ' with --bears will be debugged (even implicit dependency bears).')
 
+    devtool_exclusive_group.add_argument(
+        '--profile', nargs='?', const=True,
+        help='Enable bear profiling with cProfile. To specify where to dump the'
+        ' profiled files, supply the directory path. If specified directory'
+        ' does not exist it will be created. If the specified path points to an'
+        ' already existing file a ``FileExistsError`` is thrown . All bears'
```

Comment:
`FileExistsError` is not relevant from user perspective, just say that it's an error.
+
No space before period

Reply:
IMO it is stupid, but I was told at https://github.com/coala/coala/pull/5599#discussion_r207102115 that it was part of the agreed upon design 

A crash is never appropriate for a known problem.  A full backtrace can go to the log/console and the application should exit with an appropriately chosen exit code.

This is a simple error scenario.  how does the backtrace even help the user who choose a directory which already had profile data in it?

Reply:
>how does the backtrace even help the user who chooses a directory which already had profile data in it?
No error for this case, profiled files will get overwritten.

An error will be generated when a user try to dump profiled files in the `filepath`

Reply:
> A crash is never appropriate for a known problem. A full backtrace can go to the log/console and the application should exit with an appropriately chosen exit code.

yeah that's right. I previously thought it's working bad in coala to exit out with proper error messages, but I have just seen `get_exitcode` and that is has a branch for `SystemExit`.

@RaiVaibhav could you reraise a `SystemExit` with some error code you think that's suitable (maybe reuse an existing error code from our list), and do an error log?

Reply:
> No error for this case, profiled files will get overwritten.

That should be documented as well btw :+1:

Reply:
>That should be documented as well btw

Will create a new PR for the changes in doc